packages/api/lambdas/sf-event-sqs-to-db-records/write-granules.js

```diff
@@ -245,10 +266,30 @@ const writeGranuleAndFilesViaTransaction = async ({
   let fileRecords = [];
 
   if (workflowStatus !== 'running') {
-    fileRecords = await generateFileRecords({
-      files: updatedFiles,
-      granuleCumulusId,
-    });
+    try {
+      const filesAndErrors = await generateFileRecords({
+        files: updatedFiles,
+        granuleCumulusId,
+      });
+
+      fileRecords = filesAndErrors.files;
+
+      filesAndErrors.errors.map((e) => {
+        throw e;
+      });
+    } catch (error_) {
+      await granulePgModel.upsert(
+        trx,
+        {
+          ...granuleRecord,
+          status: 'failed',
+          error: {
+            Error: 'Failed writing files to Postgres.',
+            Cause: error_.message,
```

Comment:
We could also include ALL of the errors from `filesAndErrors.errors` here. This would just be the first one that triggered the `catch`.

Reply:
Is catching the error from the record **generation** the right approach? I would think that we want to catch errors from writing any of the files to the DB, which would fail on schema validation for missing bucket/key. Does `generateFileRecords` actually throw errors if it can't generate a bucket/key?

You would also need to refactor `writeFilesViaTransaction` so that it doesn't fail on the first failing file write. Right now it's using Promise.all so one failed file write fails them all

Reply:
I understand now why this is happening based on the logic in `translateApiFiletoPostgresFile`. See my comments below though about how I think we should change this.

Reply:
Updated this whole thing here: https://github.com/nasa/cumulus/pull/2129/commits/bf8eff5425d57c2c3b944ae7fce6b078ef29e050